rtrlib/rtr/packets.c

```diff
@@ -1080,14 +1083,38 @@ int rtr_sync_receive_and_store_pdus(struct rtr_socket *rtr_socket){
                         rtr_socket->expire_interval, rtr_socket->refresh_interval, rtr_socket->retry_interval);
             }
 
+            struct pfx_table *pfx_update_table;
+            struct spki_table *spki_update_table;
+
+            if (rtr_socket->is_resetting) {
+                RTR_DBG1("Reset in progress creating shadow table for atomic reset");
+                pfx_table_init(&pfx_shadow_table, NULL);
+                pfx_update_table = &pfx_shadow_table;
+                pfx_table_copy_except_socket(rtr_socket->pfx_table, pfx_update_table, rtr_socket);
+
+                spki_table_init(&spki_shadow_table, NULL);
+                spki_update_table = &spki_shadow_table;
+                if (spki_table_copy_except_socket(rtr_socket->spki_table, spki_update_table, rtr_socket) != SPKI_SUCCESS) {
+                    RTR_DBG1("Creation of spki shadow table failed");
+                    rtr_change_socket_state(rtr_socket, RTR_ERROR_FATAL);
+                    retval = RTR_ERROR;
+                    goto cleanup;
+                }
```

Comment:
Its a bit annoying that spki and pfx table have different code here for the same thing, but its necessary. Maybe at some point we can rework the spki table to offer a 'for_each' function and re-implement spki_table_copy_except_socket so its similar to pfx_table_copy_except_socket. The tommy library might actually offer something like this already ( a for-each function)